Automattic/jetpack
# Videopress: Request upload token with correct user

Fixes https://github.com/Automattic/wp-calypso/issues/72900

## Proposed changes:


Currently the generated token use for uploading videos does not take the user uploading the video into account but always use the blog owner as the user. This causes the blog owner to be attached as the author for every video that's uploaded. This affects audits as the activity log shows incorrect author on the video upload. See the [issue](https://github.com/Automattic/wp-calypso/issues/72900) for more information 

* Send the user requesting the upload token so that the upload token endpoint returns the correct user token.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* You will need to apply this diff D165318-code to your sandbox.
* On your Atomic site add another user B who is not the blog owner
* User [Jetpack Beta tester](https://github.com/Automattic/jetpack/pull/39796#issuecomment-2418307462) to apply this branch to your site.
* Upload a video with User B and verify that the correct author attached to the video is User B
* Do the same with User A or blog owner and verify the same
* Also verify that the activity log is showing the correct user that uploads the video

![image](https://github.com/user-attachments/assets/f2de5a29-8c5d-4312-9dfc-aef2603acbc9)
